web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -14,24 +14,69 @@
  * limitations under the License.
  */
 
-import { Component, Inject, OnInit } from '@angular/core';
+import { Component, Inject, OnDestroy } from '@angular/core';
+import { Observable } from 'rxjs';
+import { ProjectService } from '../../services/project/project.service';
 import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Project } from '../../shared/models/project.model';
+import { Layer } from '../../shared/models/layer.model';
+import { Subscription } from 'rxjs';
+import { DataStoreService } from '../../services/data-store/data-store.service';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'app-layer-dialog',
   templateUrl: './layer-dialog.component.html',
   styleUrls: ['./layer-dialog.component.css'],
 })
-export class LayerDialogComponent {
+export class LayerDialogComponent implements OnDestroy {
   layerId: string;
-
+  layer?: Layer;
+  projectId?: string;
+  activeProject$: Observable<Project>;
+  subscription: Subscription = new Subscription();
   constructor(
     // tslint:disable-next-line:no-any
     @Inject(MAT_DIALOG_DATA) data: any,
-    private dialogRef: MatDialogRef<LayerDialogComponent>
+    private dialogRef: MatDialogRef<LayerDialogComponent>,
+    private projectService: ProjectService,
+    private dataStoreService: DataStoreService,
+    private router: Router
   ) {
-    this.layerId = data.layerId!;
     // Disable closing on clicks outside of dialog.
     dialogRef.disableClose = true;
+    this.layerId = data.layerId;
+    this.activeProject$ = this.projectService.getActiveProject$();
+    this.subscription.add(
+      this.activeProject$.subscribe(project => {
+        this.layer = project.layers.get(this.layerId)!;
+        this.projectId = project.id;
+      })
+    );
+  }
+
+  onSave() {
+    if (!this.projectId) {
+      return;
+    }
+    const layerId =
```

Comment:
In that case this would move into the callback in the constructor, or even better, the subscribe lambda can be moved into a function this.onProjectLoaded(project: Project).